src/lib/coaler/embedder/ConformerEmbedder.cpp

```diff
@@ -89,4 +137,118 @@ namespace coaler::embedder {
         params.clearConfs = false;
         return params;
     }
+
+    /*----------------------------------------------------------------------------------------------------------------*/
+
+    std::vector<multialign::PoseID> ConformerEmbedder::generateNewPosesForAssemblyLigand(
+        RDKit::ROMol *worstLigandMol, const multialign::LigandVector &targets,
+        const std::unordered_map<multialign::LigandID, multialign::PoseID> &conformerIDs) {
+        std::vector<unsigned> newIds;
+        for (const multialign::Ligand &target : targets) {
+            // find mcs
+            const multialign::LigandID targetID = target.getID();
+            const multialign::PoseID targetConformerID = conformerIDs.at(targetID);
+            const RDKit::ROMol targetMol = target.getMolecule();
+            const RDKit::Conformer &targetConformer = targetMol.getConformer(targetConformerID);
+
+            std::vector<RDKit::ROMOL_SPTR> mols = {
+                boost::make_shared<RDKit::ROMol>(*worstLigandMol),
+                boost::make_shared<RDKit::ROMol>(target.getMolecule()),
+            };
+
+            auto mcsResult = get_mcs(mols, false);
+            if (mcsResult.QueryMol == nullptr) {
+                spdlog::warn("no mcs found");
+                continue;
+            }
+
+            RDKit::SubstructMatchParameters substructMatchParams;
+            substructMatchParams.uniquify = true;
+            substructMatchParams.useChirality = false;
+            substructMatchParams.useQueryQueryMatches = false;
+            substructMatchParams.maxMatches = 1;
+            substructMatchParams.numThreads = 1;
+
+            std::vector<RDKit::MatchVectType> ligandMatches;
+            RDKit::SubstructMatch(*worstLigandMol, *mcsResult.QueryMol, ligandMatches, substructMatchParams.uniquify,
```

Comment:
is there no method overload to just give this the substruct match params struct?